modules/RTC/TPCUtils.js

```diff
@@ -273,29 +274,48 @@ export class TPCUtils {
-     * initial stream encodings configuration on the sender and the resolution
-     * of the current local track added to the peerconnection.
-     * @param {MediaStreamTrack} localTrack local video track
-     * @returns {Array[number]} an array containing the resolution heights of
-     * simulcast streams configured on the video sender.
+     * Returns the calculated active state of the simulcast encodings based on the frame height requested for the send
+     * stream. All the encodings that have a resolution lower than the frame height requested will be enabled.
+     *
+     * @param {JitsiLocalTrack} localVideoTrack The local video track.
+     * @param {number} newHeight The resolution requested for the video track.
+     * @returns {Array<boolean>}
      */
-    getLocalStreamHeightConstraints(localTrack) {
-        // React-native hasn't implemented MediaStreamTrack getSettings yet.
-        if (browser.isReactNative()) {
-            return null;
-        }
-
-        const localVideoHeightConstraints = [];
-
-        // Firefox doesn't return the height of the desktop track, assume a min. height of 720.
-        const { height = 720 } = localTrack.getSettings();
+    calculateEncodingsActiveState(localVideoTrack, newHeight) {
+        const localTrack = localVideoTrack.getTrack();
+        const { height } = localTrack.getSettings();
+
+        const encodingsState = this.localStreamEncodingsConfig
+        .map(encoding => height / encoding.scaleResolutionDownBy)
+        .map((frameHeight, idx) => {
+            let active = localVideoTrack.getVideoType() === VideoType.CAMERA
+
+                // Keep the LD stream enabled even when the LD stream's resolution is higher than of the requested
+                // resolution. This can happen when camera is captured at resolutions higher than 720p but the
+                // requested resolution is 180. Since getParameters doesn't give us information about the resolutions
+                // of the simulcast encodings, we have to rely on our initial config for the simulcast streams.
+                ? newHeight > 0 && this.localStreamEncodingsConfig[idx]?.scaleResolutionDownBy === 4.0
+                    ? true
+                    : frameHeight <= newHeight
+
+                // Keep all the encodings for desktop track active.
+                : true;
+
+            // Disable the lower spatial layers for screensharing in Unified plan when low fps screensharing is in
+            // progress. There is no way to enable or disable simulcast during the call since we are re-using the same
+            // sender. Safari is an exception here since it does not send the desktop stream at all if only the high
+            // resolution stream is enabled.
+            if (this.pc.isSharingLowFpsScreen()
+                && this.pc.usesUnifiedPlan()
+                && !browser.isWebKitBased()
+                && this.localStreamEncodingsConfig[idx].scaleResolutionDownBy !== 1.0) {
```

Comment:
What happens if other layers are not disabled? Why do they need to be disabled exactly?

Reply:
For low fps sharing, i.e., when user prefers the resolution over fps, we disable the lower resolution streams since sending all 3 streams results in the browser suspending the high resolution stream in low b/w and cpu cases and we hit this more often on the low end machines. Suspending the lower streams ensures the highest resolution stream is available always.